Xamarin.Forms.Core/DefinitionCollection.cs

```diff
@@ -83,6 +83,7 @@ public void Insert(int index, T item)
 			get { return _internalList[index]; }
 			set
 			{
+				_internalList[index].SizeChanged -= OnItemSizeChanged;
```

Comment:
Please add a null + bounds check to make sure the exception thrown here when misused doesn't change.